modules/mod_sources/collection_source.ts

```diff
@@ -0,0 +1,15 @@
+//The same as the Mod-install_Source, but I had to create another, since TS does not let overriding abstract methods.
```

Comment:
I don't think you need to create this duplicate class, i want to avoid this duplication.
You should be able to use the mod source base just fine, but if needed we can make changes to make it work.

Reply:
This works, together with the array for all sources you mentioned.